source/neuropod/bindings/c/c_api.h

```diff
@@ -29,19 +29,44 @@ extern "C" {
 
 typedef struct NP_Neuropod NP_Neuropod;
 
-// Load a model given a path
+// Load a model given a path.
+// status is set if user provided a valid pointer.
```

Comment:
I'm not sure it's a good idea for status to be optional here (and in the other methods). Doing so might make this interface somewhat error prone.

It's obviously easier to pass in `nullptr` than to use and check am `NP_Status`, but I think we should force usage.

When I was working with the TF C API, it was somewhat inconvenient to have to deal with statuses everywhere, but it forced me to do error checking and I think that led to more robust user code (and a better debugging experience).

Any thoughts @selitvin

Reply:
I thought abut some extra efficiency for Infer. "error handler" instead of status may be more efficient (it would never call it on success), but let's make it required for now.